core/src/main/java/com/datastrato/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -33,4 +43,48 @@ static void checkMetalakeExists(EntityStore store, String metalake)
       throw new RuntimeException(e);
     }
   }
+
+  static RoleEntity getRoleEntity(NameIdentifier identifier) {
+
+    Cache<NameIdentifier, RoleEntity> cache =
+        GravitinoEnv.getInstance().accessControlManager().getRoleCache();
+    EntityStore store = GravitinoEnv.getInstance().entityStore();
+
+    return cache.get(
+        identifier,
+        id -> {
+          try {
+            return store.get(identifier, Entity.EntityType.ROLE, RoleEntity.class);
+          } catch (IOException ioe) {
+            LOG.error("getting roles {} failed  due to storage issues", identifier, ioe);
+            throw new RuntimeException(ioe);
+          }
+        });
+  }
```

Comment:
Why don't you do this in `RoleManager`?

Reply:
These method will be used for other managers. Maybe I can make RoleManager as other manager's field.

Reply:
Done.